staging/src/k8s.io/kubectl/pkg/cmd/create/create_clusterrole_test.go

```diff
@@ -149,26 +150,28 @@ func TestCreateClusterRole(t *testing.T) {
 	}
 
 	for name, test := range tests {
-		ioStreams, _, buf, _ := genericclioptions.NewTestIOStreams()
-		cmd := NewCmdCreateClusterRole(tf, ioStreams)
-		cmd.Flags().Set("dry-run", "true")
-		cmd.Flags().Set("output", "yaml")
-		cmd.Flags().Set("verb", test.verbs)
-		cmd.Flags().Set("resource", test.resources)
-		cmd.Flags().Set("non-resource-url", test.nonResourceURL)
-		cmd.Flags().Set("aggregation-rule", test.aggregationRule)
-		if test.resourceNames != "" {
-			cmd.Flags().Set("resource-name", test.resourceNames)
-		}
-		cmd.Run(cmd, []string{clusterRoleName})
-		actual := &rbac.ClusterRole{}
-		if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), buf.Bytes(), actual); err != nil {
-			t.Log(string(buf.Bytes()))
-			t.Fatal(err)
-		}
-		if !equality.Semantic.DeepEqual(test.expectedClusterRole, actual) {
-			t.Errorf("%s:\nexpected:\n%#v\nsaw:\n%#v", name, test.expectedClusterRole, actual)
-		}
+		t.Run(name, func(t *testing.T) {
+			ioStreams, _, buf, _ := genericclioptions.NewTestIOStreams()
+			cmd := NewCmdCreateClusterRole(tf, ioStreams)
+			cmd.Flags().Set("dry-run", "true")
+			cmd.Flags().Set("output", "yaml")
+			cmd.Flags().Set("verb", test.verbs)
+			cmd.Flags().Set("resource", test.resources)
+			cmd.Flags().Set("non-resource-url", test.nonResourceURL)
+			cmd.Flags().Set("aggregation-rule", test.aggregationRule)
+			if test.resourceNames != "" {
+				cmd.Flags().Set("resource-name", test.resourceNames)
+			}
+			cmd.Run(cmd, []string{clusterRoleName})
```

Comment:
I'd prefer we move with the testing towards the pattern from create cronjob, where you directly inject values into `*Options` struct and focus testing on the "raw meat", which in here is `RunCreateRole`.

Reply:
The less of the test factory magic, the easier this testing is for newcomers to understand it.